### ISSUE TITLE
External DB Storage

### ISSUE THREAD

**Issue by andrewmeissner** [markphelps/flipt#41]
Any plans on expanding this to use external databases, like mysql or postgres?  It's the coolest feature flag system I've seen.  :)

**Comment by markphelps** [comment]
Thanks! Glad you like it!

Yeah that is on the roadmap, my plan was to offer High Availability support (using an external database so you can run multiple copies of Flipt) as a paid feature.. but that being said.. it really shouldnt be that much work since the Go sql package _should_ allow the code to be easily adapted to multiple DB backends. So it may be something I just add to the OS version.

It would also require a change to the format of the config file since it now assumes SQLite.. but thats not a huge deal.

**Comment by andrewmeissner** [comment]
That'd be awesome!

**Comment by markphelps** [comment]
Will be tracking this in #62 so will close this. My thinking is Postgres could come first, then MySQL a close second.

Thanks again for the feature request!

**Comment by markphelps** [comment]
@andrewmeissner Postgres support just landed in v0.5.0. Please give it a try and let me know what ya think! Thanks

**Comment by andrewmeissner** [comment]
```ruby
Vagrant.configure("2") do |config|
    config.vm.define "flipt-db" do |db|
        db.vm.provider "docker" do |d|
            d.image = "postgres:12-alpine"
            d.name = "flipt-db"
            d.ports = ["5432:5432"]
            d.volumes = ["#{ENV["HOME"]}/vagrantData/flipt/db:/var/lib/postgresql/data"]
            d.env = {
                "POSTGRES_PASSWORD": "flipt",
                "POSTGRES_USER": "flipt",
                "POSTGRES_DB": "flipt"
            }
        end
    end

    (1..3).each do |i|
        config.vm.define "flipt-#{i}" do |flipt|
            flipt.vm.provider "docker" do |d|
                d.image = "markphelps/flipt:v0.6.0"
                d.name = "flipt-#{i}"
                d.ports = ["808#{i-1}:8080"]
                d.volumes = ["#{ENV["HOME"]}/vagrantData/flipt/server#{i}:/var/opt/flipt"]
                d.env = {
                    "FLIPT_DB_URL": "postgres://flipt:flipt@flipt-db:5432/flipt?sslmode=disable"
                }
                d.link("flipt-db:flipt-db")
            end
        end
    end
end
```

Works wonderfully!  Thank you so much for adding this!
